src/libs/TransactionUtils.ts

```diff
@@ -0,0 +1,359 @@
+import Onyx, {OnyxCollection} from 'react-native-onyx';
+import {format, parseISO, isValid} from 'date-fns';
+import CONST from '../CONST';
+import ONYXKEYS from '../ONYXKEYS';
+import DateUtils from './DateUtils';
+import * as NumberUtils from './NumberUtils';
+import {RecentWaypoint, ReportAction, Transaction} from '../types/onyx';
+import {Receipt, Comment, WaypointCollection} from '../types/onyx/Transaction';
+
+type AdditionalTransactionChanges = {comment?: string};
+
+type TransactionChanges = Partial<Transaction> & AdditionalTransactionChanges;
+
+let allTransactions: OnyxCollection<Transaction> = {};
+
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.TRANSACTION,
+    waitForCollectionCallback: true,
+    callback: (value) => {
+        if (!value) {
+            return;
+        }
+        allTransactions = Object.fromEntries(Object.entries({...value}).filter(([, transaction]) => !!transaction));
```

Comment:
Do we really need a copy here?
```suggestion
        allTransactions = Object.fromEntries(Object.entries(value).filter(([, transaction]) => !!transaction));
```